File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prev_map = {}

        for i, num in enumerate(nums):
            nums2 = target - num

            if nums2 in prev_map:
                return [prev_map[nums2], i]


            prev_map[num] = i 

File: 0001-two-sum/0001-two-sum.java
import java.util.Hashtable;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int [] res = new int[2];
        Hashtable<Integer, Integer> numMap = new Hashtable<>();
    
        for(int i = 0 ; i < nums.length; i++) {
            numMap.put(nums[i], i);
        }

        for(int j = 0 ; j < nums.length; j++) {
            int diff = target-nums[j];
            if (numMap.containsKey(diff) && (j != numMap.get(diff))) {
                res[0] = j;
                res[1] = numMap.get(diff);
            
            }
        }   
        return res;
    }
}